packages/skygear-core/src/client.ts

```diff
@@ -79,11 +80,21 @@ export abstract class BaseAPIClient {
     return headers;
   }
 
-  async fetch(input: string, init?: RequestInit): Promise<Response> {
+  async fetch(
+    input: string,
+    init?: RequestInit,
+    options: { autoRefreshToken?: boolean } = {}
```

Comment:
Having required argument after optional argument is weird. If we make `init` required, the function signature deviates from the original `fetch` function signature.

Reply:
`options` is not required?

Reply:
`init?` is optional

Reply:
I mean both `init` and `options` parameters are already optional.

Reply:
So they are ambiguous.

Reply:
I don't think it's ambiguous: consumer can specify `input`, `input, init`, or `input, init, options`. Most developer would not change the default options so the usage remains same as `fetch`.

Reply:
I mean the developer cannot call `input, options`

```javascript
function foo(a, b, c = "defaultC") {
  console.log("a", a);
  console.log("b", b);
  console.log("c", c);
}

console.log(`foo("a")`);
foo("a");
console.log();

console.log(`foo("a", "b")`);
foo("a", "b");
console.log();

console.log(`foo("a", "c")`);
foo("a", "c");
console.log();

console.log(`foo("a", "b", "c")`);
foo("a", "b", "c");
console.log();
```

Output
```
foo("a")
a a
b undefined
c defaultC

foo("a", "b")
a a
b b
c defaultC

foo("a", "c")
a a
b c
c defaultC

foo("a", "b", "c")
a a
b b
c c


```

Reply:
I don't think this is a problem. For a majority of developers, `options` need not to be specified. If options need to be changed, requiring developers to use `fetch("URL", undefined, { autoRefreshToken: false })` does not seems unreasonable.